CONTRIBUTING.md

```diff
@@ -75,6 +75,58 @@ If you follow that process, you can then link to the github repository in the is
 
 **NOTE**: If your issue involves a webpack config, create-react-app will prevent you from modifying the _app's_ webpack config, however you can still modify storybook's to mirror your app's version of storybook. Alternatively, use `yarn eject` in the CRA app to get a modifiable webpack config.
 
+### Executing Tests Locally
+
+Tests can be executed locally with the `yarn test` command, which gives you CLI options to execute various test suites in different modes. Some of the test suites have special set-up needs, which are listed below.
+
+The execution modes available can be selected from the cli or passed to `yarn test` with specific parameters.  Available modes include `--watch`, `--coverage`, and `--runInBand`, which will respectively run tests in watch mode, output code coverage, and run selected test suites serially in the current process.
+
+#### Core & React & Vue Tests
+
+`yarn test --core`
+
+This option executes test from `<rootdir>/app/react`, `<rootdir>/app/vue`, and `<rootdir>/lib`
+Before the tests are ran, the project must be bootstrapped with core. You can accomplish this with `yarn bootstrap --core` 
+
+#### React-Native example Tests
+
+`yarn test --reactnative`
+
+This option executes tests from `<rootdir>/app/react-native`
+Before these tests are ran, the project must be bootstrapped with the React Native example enabled.  You can accomplish this by running `yarn bootstrap --reactnative`
+
+#### Integration Tests (Screenshots of running apps)
+
+`yarn test --integration`
+
+This option executes tests from `<rootdir>/integration`
+In order for the snapshot-integration tests to be executed properly, examples being tested must be running on their defaults ports, as declared in `integration/examples.test.js`
+
+Puppeteer is used to launch and grab screenshots of example pages, while jest is used to assert matching images.
+
+### Updating Tests
+
+Before any contributes are submitted in a PR, make sure to add or update meaningful tests. A PR that has failing tests will be regarded as a “Work in Progress” and will not be merged until all tests pass.
+When creating new unit test files, the tests should adhere to a particular folder structure and naming convention, as defined below.
+
+```sh
+#Proper naming convention and structure for js tests files
++-- parentFolder
+|   +-- [filename].js
+|   +-- [filename].test.js
+```
+
+#### Updating Integration Screenshots
+
+Integration screenshots can be updated using pupeteer's screenshot command. For example:
```

Comment:
I believe this can be done easier, using `jest -u`. Is it true @ndelangen ?

In `release/3.3`, I'v added an `--update` option to `yarn test`. Maybe we should back-port it into `master`

Reply:
I've cut out that bit in favor of the e2d update instructions in 2b